### DISCUSSION TITLE
Feed Plugin Sorting

### DISCUSSION THREAD

**Discussion by fdevans** [vuepress-theme-hope/vuepress-theme-hope#3487] [Q&A] [answered]
We are having a tough time understanding how to sort using the Feed Plugin.  Hoping someone has some examples.

We tried brining in and rewriting the CompareDate function from the plugin:

```
const compareDate = (
  dateA: Date | string | undefined,
  dateB: Date | string | undefined,
): number => {
  if (!dateA || !(dateA instanceof Date)) return 1;
  if (!dateB || !(dateB instanceof Date)) return -1;
  return dateB.getTime() - dateA.getTime();
};
```

and then tried this `sort: ({ pageA }: Page, {pageB}: Page): number => compareDate(pageA.frontmatter.date, pageB.frontmatter.date)`

We are still seeing content ordered randomly.

Our goal is to sort, descending, purely on the frontmatter.date.

Anyone have working examples of custom sorts?

**Comment by Mister-Hope** [comment]
China is in national celebration vacation till 7th, your discussion will be handled later, so will others.

**Reply by Mister-Hope** [reply]
I will start handling your issues and discussions now,

**Comment by fdevans** [comment]
No problem, enjoy.  🥳

**Comment by Mister-Hope** [comment]
I think https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/18377386a088134fefbe2a5729b16309f6b72af6 shall fix this, I will release a new version probably in 24h

**Reply by fdevans** [reply]
Thank you for the refactor, but we are still unclear what the `sort` option should include.  Are these docs still valid?
https://theme-hope.vuejs.press/config/plugins/feed.html#sort

I tried the default by leaving sort blank, and the method I described above, both return the same un-ordered results.

If there is an example how we'd sort by date the following frontmatter entries it would be helpful.

```
---
date: 2023-09-23
feed:
  description: "List at top"

---
```

```
---
date: 2023-09-01
feed:
  description: "List 2nd"

---
```

**Reply by fdevans** [reply, marked as answer]
OK ... I fixed this with the fixed from version 238 mentioned above.

The `sort` entry needs to be called `sorter` and now the following entry properly sorts on `frontmatter.date` (docs are wrong, I'll try to send a PR): 

```
      feed: {
        hostname: 'https://yourhost.com',
        rss: true,
        json: true,
        filter: ({ frontmatter, filePathRelative }: Page): boolean => !(frontmatter.feed === undefined || frontmatter.home || !filePathRelative || frontmatter.article === false || frontmatter.feed === false),
        sorter: (
          pageA: Page,
          pageB: Page,
        ): number =>
          compareDate( pageA.frontmatter.date, pageB.frontmatter.date)
      }
```

Also important to note that the following import statements needed to be updated:

`import { defineUserConfig, Page } from 'vuepress';`  This one needed the "Page" added to it.
`import { compareDate } from "vuepress-shared/node";` This one uses the new `compareDate` shared util.  (also had to be added as a package using `npm install -D vuepress-shared`)
